libs/base/Data/Vect.idr

```diff
@@ -361,6 +361,34 @@ mapMaybe f (x::xs) =
        Just y  => (S len ** y :: ys)
        Nothing => (  len **      ys)
 
+||| Map with index parameter
+|||
+||| ```idris example
+||| mapI (*) [4,1,2,1]
+||| ```
+export
+mapI : (Nat -> a -> b) -> Vect len a -> Vect len b
```

Comment:
Why using `Nat` index for `Vect`s? (except for the tail recursion, maybe) I'm sure `Fin len` is much more usable for `Vect` (I personally find `Fin`-based `mapI` even for `List`s in my codebase).

Reply:
yea I already noticed that. (I'm realy new to idris, maybe not best idea to contribute than) I alredy chnaged alll Vect functions, and put them in the contrib library, but now I have a problem being in HEAD detached in git, I'm just looking up the solution for that